src/login-form/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { theme, ThemeProperties } from '@dojo/framework/core/middleware/theme';
+import { i18n } from '@dojo/framework/core/middleware/i18n';
+import Button from '../button';
+import TextInput from '../text-input';
+import PasswordInput from '../password-input';
+import Form, { FormProperties } from '../form';
+import * as css from '../theme/default/login-form.m.css';
+import bundle from './nls';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+
+export interface LoginFormFields {
+	username: string;
+	password: string;
+}
+
+interface BaseFormProperties extends ThemeProperties {
+	/* The initial value of the login form */
+	initialValue?: Partial<LoginFormFields>;
+
+	onSubmit?: never;
+	action?: never;
+}
+
+type Omit<T, E> = Pick<T, Exclude<keyof T, E>>;
+
+interface SubmitProperties extends Omit<BaseFormProperties, 'onSubmit'> {
+	/** Callback for when the form is submitted with valid values */
+	onSubmit(values: LoginFormFields): void;
+}
+
+interface ActionProperties extends Omit<BaseFormProperties, 'action'> {
+	/** Action url for the form on submit */
+	action: string;
+	/** method of submit, defaults to `post` */
+	method?: 'post' | 'get';
+}
+
+export type LoginFormChildren =
+	| undefined
+	| {
+			forgotPassword?(): RenderResult;
+			register?(): RenderResult;
+	  };
+
+export type LoginFormProperties = SubmitProperties | ActionProperties;
+
+function isSubmitForm(properties: LoginFormProperties): properties is SubmitProperties {
+	return (properties as SubmitProperties).onSubmit !== undefined;
+}
+
+function isActionForm(properties: LoginFormProperties): properties is ActionProperties {
+	return (properties as ActionProperties).action !== undefined;
+}
+
+const factory = create({ theme, i18n })
+	.properties<LoginFormProperties>()
+	.children<LoginFormChildren>();
+
+const LoginForm = factory(function LoginForm({
+	properties,
+	children,
+	middleware: { theme, i18n }
+}) {
+	const classes = theme.classes(css);
+
+	let formProps: FormProperties = {};
+	const props = properties();
+
+	if (isSubmitForm(props)) {
+		const { onSubmit, initialValue } = props;
+		formProps = {
+			onSubmit,
+			initialValue
+		};
+	} else if (isActionForm(props)) {
+		const { action, initialValue, method } = props;
+		formProps = {
+			action,
+			method,
+			initialValue
+		};
+	}
+
+	const [
+		{ forgotPassword, register } = { forgotPassword: undefined, register: undefined }
+	] = children();
+
+	const { messages } = i18n.localize(bundle);
```

Comment:
How does a user provide additional nls? perhaps an example would be good...

Reply:
do we have examples of this elsewhere?